flexget/plugins/input/json.py

```diff
@@ -0,0 +1,137 @@
+"""Plugin for json file."""
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *  # noqa pylint: disable=unused-import, redefined-builtin
+import dateutil.parser as parser
+
+import re
+import os
+import logging
+import glob
+
+from flexget import plugin
+from flexget.utils import json
+from flexget.entry import Entry
+from flexget.event import event
+
+log = logging.getLogger('json')
+
+
+class Json(object):
+    """
+    Parse a json file for entries using regular expression.
+
+    :: 
+
+      files: <path to JSON file(s)>
+      encoding: <JSON encoding>
+      entry:
+        - <field>: <corresponding JSON key>
+        - <field>
+        - get_remnants
+
+    Note: each entry must have at least two fields, 'title' and 'url'. If not specified in the config, 
+    this plugin asssumes that keys named 'title' and 'url' exist within the JSON.
+
+    Example::
+
+      json:
+        files: entries.json
+        encoding: utf8
+        entry:
+          - title: name
+          - url: 'web_.*'
+          - get_remnants
+          
+    """
+    
+    schema = {
+        'type': 'object',
+        'properties': {
+            'files': {'type': 'string'},
+            'encoding': {'type': 'string', 'default': 'utf-8'},
+            'entry': {
+                'type': 'array',
+                'items': {
+                    'oneOf': [
+                        {'type': 'string'},
+                        {
+                            'type': 'object',
+                            'additionalProperties': {
+                                'type': 'string', 'format': 'regex'
+                            },
+                        },
+                    ]
+                },
+            },
+        },
+        'required': ['files'],
+        'additionalProperties': False,
+    }
+                    
+    def dt(val):
+        try:
+            return parser.parse(val)
+        except (ValueError, OverflowError):
+            return val
+
+    def on_task_input(self, task, config):
+        files = os.path.expanduser(config['files'])
+        json_encoding = config['encoding']
+        entry_config = config.get('entry')
+        
+        fields = {}
+        get_remnants = False
+        if config.get('entry'):
+            for required_field in config.get('entry'):
+                if isinstance(required_field, str):
+                    if required_field == 'get_remnants':
+                        get_remnants = True
+                    else:
+                        fields[required_field] = re.compile('^' + required_field + '$')
+                else:
+                    fields[next(iter(required_field))] = re.compile(
```

Comment:
1. I'm not sure why you're trying to get the first item like that, could just do `required_field[0]`, can't you?
2. Passing that value to a variable would reduce some resusage of that slightly less than readable code. You could use fstring as well:

```python
value = next(iter(required_field))
fields[value] = f'^{value}$'
```

Reply:
I tried the first suggestion, but it threw a `KeyError`. The second part was a good suggestion!